table/block_based_table_reader.cc

```diff
@@ -2738,7 +2741,9 @@ Status BlockBasedTable::Get(const ReadOptions& read_options, const Slice& key,
         }
 
         bool may_exist = biter.SeekForGet(key);
-        if (!may_exist) {
+        // If user-specified timestamp is supported, we cannot end the search
```

Comment:
This is the last to consider but the hash-based data block might need some attention there.